invenio_records_resources/services/records/service.py

```diff
@@ -579,3 +581,74 @@ def on_relation_update(
 
             self.reindex(identity, search_query=search_query)
         return True
+
+    @unit_of_work()
+    def create_or_update_many(self, identity, data, uow=None):
+        """Create or update a list of records.
+
+        This method takes a list of record data and creates or updates the corresponding records.
+
+        Args:
+            identity (object): The user identity performing the operation.
+            data (list): A list of tuples containing the record ID and record data.
+            uow (UnitOfWork, optional): The unit of work to register the record operations. Defaults to None.
+
+        Returns:
+            list: A list of tuples containing the operation type ('create' or 'update'), the processed record or the record dict, and any schema errors encountered.
+        """
+        records_processed = []
+        for record_id, record_dict in data:
+            try:
+                record = self.record_cls.pid.resolve(record_id)
+
+                # Permissions
+                self.require_permission(identity, "update", record=record)
+                record_data, schema_errors = self.schema.load(
+                    record_dict,
+                    context=dict(identity=identity, pid=record.pid, record=record),
+                    raise_errors=False,
+                )
+
+                # If errors we avoid creating the record
+                if schema_errors:
+                    records_processed.append(("update", record_dict, schema_errors))
```

Comment:
```suggestion
                    record = self.recordDTO(record_dict)
                    records_processed.append(("update", record_dict, schema_errors))
```

When it fails we are returning the record dict but on success we return the record class, should we try to wrap in a DTO class to be consistent with what we return? or are we good with this?

Actually this might be a bigger change to implement, just sharing my thoughts in case you guys have better ideas for this.

Reply:
my 2 cents: as someone using those methods, i would prefer to be consistent what will be returned. it is already irritating that no RecordItem will be returned as from the other methods.

Reply:
could you explain what DTO expands to? I am missing some context, and in general, it would be nice not to use abbreviations in the method names

Reply:
Sure, DTO stands for Data Object Transfer. It would allow us to avoid returning some times a dict class of the record -when it fails - and a Record class - when it succeded. The DTO object would take care of how to access the data. However another simpler alternative is to simply return a dict class of the record all the time.

Reply:
I would be very hesitant to write another method just to deal with inconsistency of what we return. It does not solve the problem, just hides it.

Reply:
Addressed with new `ServiceBulkListResult`